### PR TITLE
Adjust to PackageCollectionSigning changes

### PR DESCRIPTION
**Pull request by yim-lee**
https://github.com/apple/swift-package-manager/pull/6609

### CODE REVIEW

**Comment by yim-lee** [comment]
Ran tests locally for Swift 5.7, 5.8, 5.9 and nightly.

Tested `package-collection-sign` on macOS and Ubuntu Jammy.

**Comment by yim-lee** [comment]
Also tried signing SSWG collection with these changes and it worked.

File: Package.swift [hunk 1]
```diff
@@ -12,8 +12,8 @@ let package = Package(
         .executable(name: "package-collection-diff", targets: ["PackageCollectionDiff"]),
     ],
     dependencies: [
-        .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMajor(from: "1.0.2")),
-        .package(url: "https://github.com/apple/swift-package-manager.git", branch: "main"),
+        .package(url: "https://github.com/apple/swift-argument-parser.git", .upToNextMajor(from: "1.2.2")),
+        .package(url: "https://github.com/yim-lee/swift-package-manager.git", branch: "async-collection-signer"),
```
Comment (tomerd) [inline Package.swift:16]: 👀

**Review by tomerd — APPROVED** [review]
lgtm, just the dependency needs to change once the SwiftPM PR is merged

**Pull request merged**